Documentation/building/build-guidelines.md

```diff
@@ -0,0 +1,47 @@
+Build guidelines
+================
+###Goal
+
+It is difficult for devs, who are new to the corefx repo, to reason about the build and understand how to implement new features, or otherwise update the repo's build.  The primary goal is to provide a repo which is easier to reason about.  A secondary goal, would be to help the dev implement their changes to the build (without getting in their way) such that they are confident they are following our guidelines.
+
+- Keep it simple
+- Keep it clean
+- Make it maintainable (secondary)
+
+### Keep it simple
+
+We don't want multiple ways to do the same thing.  There is a support cost and mind-share cost associated with having multiple ways to achieve something.  It's ok to have insider tricks, things for power-users who understand the default experience but want to optimize it for their development patterns.
+- Support cost
+ - Discovery becomes difficult
+ - Maintenance is difficult
+ - Devs will grow accustomed to whatever pattern they first learn and not utilize the other, so why have more than one?
+ - Multiple ways to accomplish the same goal make it difficult to ensure that what a dev is doing is the same thing that an automated system is doing (or close to it)
+- If the second way to do something is literally an alias to the first way, then that's probably ok, but we should have those things in a discoverable place, not cluttering our root.
+  - The run tool can highlight an alternative command, ie. Build-managed.cmd, build-tests.cmd, etc…
+
+### Keep it clean
+
+It is difficult to understand our build because our build files are interspersed throughout our repo, or simply dropped at the root where they are lost amongst the myriad of other files in the root of the repo.  It is preferable if there is a single place to go, to reason about our build system.  
```

Comment:
Yes, I agree that it would make sense to move many of these (.builds, some of the .proj files, etc...) along with the others.  I'd want to leave the "structural" ones such as dir.props, dir.targets, dirs.proj (maybe).  dir.props could probably be eliminated, but I think it's fine to leave dir.props / dir.targets files.